test/StringClient.test.js

```diff
@@ -59,6 +71,39 @@ describe('StringClient', function () {
         });
     });
 
+    it('should filter out undefined headers', function (done) {
```

Comment:
Can you also add a test for constructor time headers?

Reply:
it's already passed to the constructor